src/rand.rs

```diff
@@ -93,7 +93,7 @@ impl SecureRandom for SystemRandom {
 
 impl private::Sealed for SystemRandom {}
 
-#[cfg(not(any(target_os = "linux", target_os = "macos", target_os = "ios", windows)))]
+#[cfg(not(any(target_os = "linux", target_os = "macos", target_os = "ios", windows, target_env = "sgx")))]
```

Comment:
Maybe just changes this to: `not(all(unix, any(...)))`. Why would urandom be available?